v2client/algod/v2wrappers.go

```diff
@@ -0,0 +1,116 @@
+package algod
+
+import (
+	"context"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/types"
+	"github.com/algorand/go-algorand-sdk/v2client/common"
+	"github.com/algorand/go-algorand-sdk/v2client/common/models"
+)
+
+// TODO ejr received peer feedback to have Block return types.Block not models.Block?
+
+func (client Client) Shutdown(ctx context.Context, timeout models.ShutdownParams, headers ...*common.Header) error {
+	return client.post(ctx, nil, "/shutdown", timeout, headers)
+}
+
+func (client Client) RegisterParticipationKeys(ctx context.Context, account string, params models.RegisterParticipationKeysAccountIdParams, headers ...*common.Header) error {
+	return client.post(ctx, nil, fmt.Sprintf("/register-participation-keys/%s", account), nil, headers)
+}
+
+func (client Client) PendingTransactionInformation(ctx context.Context, txid string, params models.GetPendingTransactionsParams, headers ...*common.Header) (result types.Transaction, err error) {
```

Comment:
it returns `SignedTransaction` one and the response actually contains more information than that. @winder can elaborate.

Reply:
@rotemh @winder Do I follow correctly: 
- this method should return `types.SignedTransaction`
- the actual client will internally receive a `models.Transaction` from the server - https://github.com/algorand/indexer/blob/master/api/generated/types.go#L307 - but that has the information in it needed to rebuild the `types.SignedTransaction`.

Reply:
if you ask for the `format=msgpack`, it should be already in `types.SignedTransaction` and you can just unmarshall it. The node will return it as it has it.

Reply:
Evan take a look here:
https://algo-indexer-v2.docs.stoplight.io/reference/v2-transactions-pending-txid/pendingtransactioninformation?__build_id=58434

You can see additional data included (pool error, some round information, etc).

The SDKs are responsible for converting the `txn` field into a `SignedTransaction`